src/nus/cs2103/addressbook/AddressBook.java

```diff
@@ -30,188 +39,258 @@
-    public static final String COMMAND_HELP_EXAMPLE = COMMAND_WORD_HELP;
+    private static final String COMMAND_WORD_HELP = "help";
+    private static final String COMMAND_HELP_DESC = "Shows program usage instructions.";
+    private static final String COMMAND_HELP_EXAMPLE = COMMAND_WORD_HELP;
 
-    public static final String COMMAND_WORD_EXIT = "exit";
-    public static final String COMMAND_EXIT_DESC = "Exit the program.";
-    public static final String COMMAND_EXIT_EXAMPLE = COMMAND_WORD_EXIT;
+    private static final String COMMAND_WORD_EXIT = "exit";
+    private static final String COMMAND_EXIT_DESC = "Exit the program.";
+    private static final String COMMAND_EXIT_EXAMPLE = COMMAND_WORD_EXIT;
 
 
-    // These are ordering indexes for the different data parameters of a person.
-    // Used by the internal person String[] storage format
-    public static final int PERSON_DATA_INDEX_NAME = 0;
-    public static final int PERSON_DATA_INDEX_PHONE = 1;
-    public static final int PERSON_DATA_INDEX_EMAIL = 2;
-    public static final int PERSON_DATA_COUNT = 3; // 3 types of internal data for a person
+    /* We use a String array to store details of a single person.
+     * The constants given below are the indexes for the different data elements of a person
+     * used by the internal String[] storage format.
+     * For example, a person's name is stored as the 0th element in the array.
+     */
+    private static final int PERSON_DATA_INDEX_NAME = 0;
+    private static final int PERSON_DATA_INDEX_PHONE = 1;
+    private static final int PERSON_DATA_INDEX_EMAIL = 2;
 
-    // Listings displayed to the user use 1-indexing, but the internal model uses 0-indexing
-    public static final int DISPLAYED_INDEX_OFFSET = 1;
+    /**
+     * The number of data elements for a single person.
+     */
+    private static final int PERSON_DATA_COUNT = 3;
 
-    // This filename is used if the user doesn't provide the file name
-	public static final String DEFAULT_STORAGE_FILEPATH = "storage.txt";
-    // If the first non-whitespace character in a user's input line is this, that line will be ignored.
-    public static final char INPUT_COMMENT_MARKER = '#';
+    /**
+     * Offset required to convert between 1-indexing and 0-indexing.COMMAND_
+     */
+    private static final int DISPLAYED_INDEX_OFFSET = 1;
+
+
+
+    /**
+     * If the first non-whitespace character in a user's input line is this, that line will be ignored.
+     */
+    private static final char INPUT_COMMENT_MARKER = '#';
 
     /*
      * This variable is declared for the whole class (instead of declaring it
      * inside the readUserCommand() method to facilitate automated testing using
      * the I/O redirection technique. If not, only the first line of the input
-     * text file will be processed. TODO FIND OUT WHY
+     * text file will be processed.
      */
     private static final Scanner SCANNER = new Scanner(System.in);
```

Comment:
There are lot of other data that are declared at class level so trying to protect scanner specifically doesn't make much sense. Putting it in the main method hurts the abstraction level of the main method (as scanner is a low level detail). Ideally, it should hidden inside the readUserInput method but we can't do that because it will not work for our test technique.